carapace-server/src/main/java/org/carapaceproxy/api/CertificatesResource.java

```diff
@@ -183,4 +199,33 @@ static String stateToStatusString(DynamicCertificateState state) {
                 return "unknown";
         }
     }
+
+    @POST
+    @Path("{domain}/upload")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response uploadCertificate(@PathParam("domain") String domain,
+            @FormDataParam("file") InputStream uploadedInputStream) throws Exception {
+
+        HttpProxyServer server = (HttpProxyServer) context.getAttribute("server");
+        ConfigurationStore store = server.getDynamicConfigurationStore();        
+
+        // Salvo il certificato su db
+        String encodedChain = Base64.getEncoder().encodeToString(uploadedInputStream.readAllBytes());
+        CertificateData cert = new CertificateData(domain, "", encodedChain, AVAILABLE.name(), "", "", true);
+        store.saveCertificate(cert);       
```

Comment:
this stuff must be in HttpServer:
- save certificate
- update configuration
- apply configuration

from line 215 to line 226